llvm-general/src/LLVM/General/Internal/Attribute.hs

```diff
@@ -58,7 +65,13 @@ instance Monad m => EncodeM m A.FA.FunctionAttribute (Ptr FFI.FunctionAttrBuilde
     liftIO $ FFI.attrBuilderAddStringAttribute b kindP kindLen valueP valueLen
   encodeM a = return $ \b -> liftIO $ case a of
     A.FA.StackAlignment v -> FFI.attrBuilderAddStackAlignment b v
+    A.FA.AllocSize x y -> case y of
+                            Nothing -> FFI.attrBuilderAddAllocSize b x 0 0
+                            Just y' -> FFI.attrBuilderAddAllocSize b x 1 y'
```

Comment:
Certainly. Haskell isn't that much else, after all. The case would wind up in the encode instance in Coding.hs (where it would be reusable by for any other FFI function which winds up with a similar signature.